src/components/select/select-textbox/select-textbox.component.js

```diff
@@ -116,13 +118,16 @@ const SelectTextbox = React.forwardRef(
     function getInputAriaAttributes() {
       const joinIds = (...ids) =>
         ids.filter((item) => item !== undefined).join(" ");
-      const ariaLabelledby = hasTextCursor
-        ? joinIds(labelId, accessibilityLabelId)
-        : joinIds(labelId, textId.current);
+      const combinedAriaLabelledBy = hasTextCursor
+        ? joinIds(
+            ariaLabelledBy || labelId,
+            accessibilityLabelId || textId.current
+          )
+        : joinIds(ariaLabelledBy || labelId, textId.current);
 
       return {
         "aria-expanded": readOnly ? undefined : isOpen,
-        "aria-labelledby": ariaLabelledby || undefined,
+        "aria-labelledby": ariaLabel ? ariaLabelledBy : combinedAriaLabelledBy,
```

Comment:
I've spent a ridiculously long time thinking about this line - I found it very confusing to understand, and on further thought I don't think it's correct in the edge case of both `ariaLabel` and `ariaLabelledBy` props being provided by the consumer.

I think the correct logic is `(ariaLabel && !ariaLabelledBy) ? undefined : combinedAriaLabelledBy` - this is only different from yours in the case where both props are provided, which should be rare if it happens at all, but in that case surely we should use the combined label (which already takes `ariaLabelledBy` into account). I also think this makes the logic clearer - use the computed "combined label" except when an `ariaLabel` prop is provided and not overridden by `ariaLabelledBy`.

Reply:
my idea was to allow users to completely overwrite the `aria-labelledby` attribute and let them create their own descriptions. I can see that this might be a bit confusing, so I'll just change that to have the combined id's only.